lib/route.coffee

```diff
@@ -163,7 +163,9 @@ class @Route
     @returns False if authentication fails, and true otherwise
   ###
   _authAccepted: (endpointContext, endpoint) ->
-    if endpoint.authRequired
+    if endpoint.authOptional
+      @_authenticate endpointContext, true
+    else if endpoint.authRequired
```

Comment:
The check for the `authRequired` should go first. That way, if someone accidentally declares an endpoint as both `authRequired` and `authOptional`, the auth requirement takes precedence. Just a security check.